### PR TITLE
DX: remove 'create' method in internal classes

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: src/FileReader.php [hunk 1]
```diff
@@ -38,11 +33,13 @@ final class FileReader
      */
     public static function createSingleton()
     {
-        if (null === self::$instance) {
-            self::$instance = new self();
+        static $instance = null;
+
+        if (!$instance) {
```
Comment (kubawerlos) [inline src/FileReader.php:38]:
```suggestion
        if (null === $instance) {
```

File: src/FileReader.php [hunk 2]
```diff
@@ -38,11 +33,13 @@ final class FileReader
      */
     public static function createSingleton()
```
Comment (kubawerlos) [inline src/FileReader.php:34]: What is the reason this methods stays?

File: src/Tokenizer/Transformers.php [hunk 3]
```diff
@@ -47,7 +47,7 @@ private function __construct()
     /**
      * @return Transformers
      */
-    public static function create()
+    public static function createSingleton()
```
Comment (kubawerlos) [inline src/Tokenizer/Transformers.php:50]: What is the motivation behind rename? Not sure the new one is anyhow better.

**Review by kubawerlos — CHANGES_REQUESTED** [review]

File: src/FileReader.php (hunk 1 above)
Comment (keradus) [reply src/FileReader.php:38]: sorry @kubawerlos , i clicked "merge" and only after that I your comment popped up

File: src/Tokenizer/Transformers.php (hunk 3 above)
Comment (keradus) [reply src/Tokenizer/Transformers.php:50]: to unify with `FileReader::createSingleton`

File: src/FileReader.php (hunk 2 above)
Comment (keradus) [reply src/FileReader.php:34]: I was not focusing on having or not a need behind this method. I found a Singleton creator and unified the implementation with other class

File: src/FileReader.php (hunk 1 above)
Comment (kubawerlos) [reply src/FileReader.php:38]: @keradus make an another PR then😉

File: src/Tokenizer/Transformers.php (hunk 3 above)
Comment (kubawerlos) [reply src/Tokenizer/Transformers.php:50]: Then why not renaming in the `FileReader`? - I believe in OS `create` is most common way for creating class instance.

File: src/FileReader.php (hunk 1 above)
Comment (keradus) [reply src/FileReader.php:38]: actually, there is no rule enforcing us to use any of the approaches. until we have the strong rule/check, enforcing it, we will always diverge. I see no value in raising PR proposing such change for a single place :shrug:

File: src/Tokenizer/Transformers.php (hunk 3 above)
Comment (keradus) [reply src/Tokenizer/Transformers.php:50]: we used `create` for creating an instance and `createSingleton` for creating singleton instance <- better visibility of hacky instance creation

**Pull request merged**